### ISSUE TITLE
Offsets incorrectly displaying in annotation

### ISSUE THREAD

**Issue by Lolologist** [heartexlabs/label-studio#1008]
![image](https://user-images.githubusercontent.com/1494871/120522496-d3c1b780-c39a-11eb-837b-f2481518dcb7.png)

![image](https://user-images.githubusercontent.com/1494871/120522781-e6d48780-c39a-11eb-82f7-a4fc35d905b9.png)

In the attached pictures you can see the proper text for "Ecuador" as an entity, and the annotation tool showing it incorrectly.
My first guess is that it is related to the emoji being used, but if so, it isn't all emoji:
![image](https://user-images.githubusercontent.com/1494871/120523073-1c797080-c39b-11eb-8e48-2f3d6d64d55c.png)
As you can see in the third image, offsets continue correctly in that particular case after being exposed to some emoji.

**Comment by Lolologist** [comment]
the text before the incorrect Ecuador:
"Network data from the NetBlocks internet observatory show that mobile network operator Claro has suffered severe outages beginning 20:00 UTC (3 p.m. local time), Saturday 12 October 2019, lasting for several hours. Confirmed: Severe disruption to #Claro mobile internet service across #Quito from ~20:00 UTC (3 p.m. local time) as #Ecuador crisis escalates 📉📰 https://t.co/2ExbFY54zq pic.twitter.com/6P13WjAkuW— NetBlocks (@netblocks) October 13, 2019The outages have been widely reported by customers and come as the Army has been deployed in "
I checked manually in the source for the task and the offsets are still correct there. I expect this is an issue with the visualization part only.

**Comment by Lolologist** [comment]
Manually removing the incorrect offset and adding what *appears* in the annotator as "Ecuador" (see picture) results in incorrect offsets, now off by 2 (as might be expected): 
![image](https://user-images.githubusercontent.com/1494871/120523647-be995880-c39b-11eb-81c4-26eaadea6a2b.png)

**Comment by Lolologist** [comment]
![image](https://user-images.githubusercontent.com/1494871/120523831-0324f400-c39c-11eb-8c41-431d5810e74a.png)
offset for "crisis escalates" is correct, https should be 360, not 362.
It seems these specific emoji are being counted as two characters long.
Both of the two emoji here were introduced as part of Unicode 6.0 and documented as part of Emoji 1.0, so they aren't recent.

**Comment by makseq** [comment]
Yes, these problems usually happen because of emoji and special characters.

**Comment by lluissalord** [comment]
Is there any fix in process to correctly count the characters of emoji and special characters to not have this issue of offsets?

**Comment by makseq** [comment]
@lluissalord what LS version do you have now?

**Comment by lluissalord** [comment]
@makseq I am using version 1.3

An example of what is happening to me is below, where "Jr" should be labelled as SENIORITY. The current result for this case is:
```
{
          "value": {
            "labels": [
              "SENIORITY"
            ],
            "start": 21,
            "end": 23,
            "text": "Jr"
          },
          "id": "45",
          "from_name": "label",
          "to_name": "text",
          "type": "labels"
}
```
![image](https://user-images.githubusercontent.com/7021552/134132234-b45ae7fc-43e9-4227-b36f-ce6ffeb903d4.png)

Hence, we can see that the value on the result is correct, however the visualization on Label Studio is not correct. 

Besides, if I try to label it correctly, now the "start" and "end" does not match to what it should counting the characters:
```
{
          "value": {
            "start": 23,
            "end": 25,
            "text": "Jr",
            "labels": [
              "SENIORITY"
            ]
          },
          "id": "9StoKHXy__",
          "from_name": "label",
          "to_name": "text",
          "type": "labels"
}
```

**Comment by nicholasrq** [comment]
@lluissalord could you please share a sample text, your labeling config and a full result that's displaying incorrectly? it'd be super helpful for debugging the issue

**Comment by lluissalord** [comment]
Sample text: "👋🏽 Hola, soy Roberto Jr"

Labelling config:
```
<View>
  <Labels name="label" toName="text">
    <Label value="DATE"/>
    <Label value="LANGUAGE"/>
    <Label value="SENIORITY"/>
  </Labels>
  <Text name="text" value="$text"/>
</View>
```

Result:
```
"result": [
        {
          "value": {
            "labels": [
              "SENIORITY"
            ],
            "start": 21,
            "end": 23,
            "text": "Jr"
          },
          "id": "0",
          "from_name": "label",
          "to_name": "text",
          "type": "labels"
        }
      ],
      "was_cancelled": false,
      "ground_truth": false,
      "created_at": "2021-09-21T10:27:33.810925Z",
      "updated_at": "2021-09-21T10:27:33.810959Z",
      "lead_time": null,
      "task": 20075
    }
  ]
```

**Comment by nicholasrq** [comment]
thank you! i'll get back with further investigation

**Comment by hlomzik** [comment]
Hi all! @Lolologist @lluissalord 
I fixed this some time ago, but forgot to update LS with this fix. Could you please test this PR — https://github.com/heartexlabs/label-studio/pull/1559? My manual and automatic tests are fine on it, so just want to check if it fixes your problems.
Thanks!

**Comment by lluissalord** [comment]
Hi @hlomzik I can confirm that the PR fixed my use case. Thank you!!

**Comment by lluissalord** [comment]
Hi @nicholasrq @makseq Do we know when will be PR from @hlomzik (#1559) be included on master?

**Comment by JulesBelveze** [comment]
Currently having the same problem with v1.4, any idea when this fix is expected to be released?

**Comment by makseq** [comment]
@JulesBelveze Are you on master branch from LS github repository?

**Comment by JulesBelveze** [comment]
Nop, I'm on v1.4.0. I was just wondering if there's a patch release expected soon

**Comment by makseq** [comment]
Yes, we are going to release 1.4.1 include this patch too.
